azext_iot/iothub/params.py

```diff
@@ -577,10 +566,39 @@ def load_iothub_arguments(self, _):
             "{YYYY}. Any one placeholder may be specified at most once, but order and non-placeholder components are "
             "arbitrary. If partition key name is provided, partition key template defaults to {deviceid}-{YYYY}-{MM}",
         )
+
+    for endpoint_type in ["cosmosdb-container", "storage-container"]:
+        for endpoint_op in ["create", "update"]:
+            with self.argument_context(f"iot hub message-endpoint {endpoint_op} {endpoint_type}") as context:
+                context.argument(
+                    "endpoint_account_name",
+                    options_list=["--endpoint-account"],
+                    help="The account name for the endpoint resource.",
+                )
+
+    with self.argument_context("iot hub message-endpoint update") as context:
         context.argument(
-            "endpoint_account_name",
-            options_list=["--endpoint-account"],
-            help="The account name for the endpoint resource.",
+            "connection_string",
+            options_list=["--connection-string", "-c"],
+            help="Connection string of the routing endpoint. Use `update` if you want the connection string to "
+            "be fetched from the service instead of providing it in the argument.",
```

Comment:
Note that `update` is for "I want the CLI to find the new connection string for me. 

This is difficult since we can get sources for the connection string from different places:
1. the user inputs (which I currently enforce) - similar to create
2. the existing connection string
3. the existing properties of the endpoint (if no connection string)

We could also say if you want to update the connection string, you must provide the connection string yourself (and skip the update)

Reply:
Also note that if we do 2 and 3, we would have to decide what we would want to do if a person gives some user input that is used to get the connection string. It should win over the existing properties.

Reply:
let me check again if there is a way to have it as a flag. It would have to be something so that we know if the user wants us to update the connection string or not touch it at all

I was going to have a separate arg but decided it is prob better to keep one arg and I followed the update pattern from identity [system]

Yes, the update value means I want to update the connection string but I want you to get it for me

Reply:
There is a way
![image](https://github.com/Azure/azure-iot-cli-extension/assets/73560279/5b18911b-4cc1-487c-a294-99e7798ce6ab)

but the more I think about it, the more I want to keep it as force the user to type in "update" to show that they want us to update the connection string (which would hold more weight if the identity type is changed)

Reply:
After talking offline with Ryan, I decided to just make this only take in the connection string. So the user must provide the connection string (and we would not look it up for them)